### PR TITLE
Allow returning nulls for nullable boolean columns

### PR DESCRIPTION
**Pull request by eyalhakim**

### CODE REVIEW

**Comment by eyalhakim** [comment]
@pleerock new and clean pr here

**Comment by eyalhakim** [comment]
@pleerock these tests fails also without my changes

**Comment by pleerock** [comment]
they are failing on master because probably my latest changes. just skip failing tests for now I'll fix them on monday. write your own tests for now. you can use `it.skip(...` and `it.only(..` to skip or run only yours

**Comment by eyalhakim** [comment]
@pleerock done

File: test/functional/driver/convert-to-entity/convert-to-entity.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    before(async () => connections = await createTestingConnections({
+        entities: [Post],
+        schemaCreate: true,
+        dropSchemaOnConnection: true
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it.only("should return null value in entity property when record column is null", () => Promise.all(connections.map(async connection => {
+        const userRepository = connection.getRepository<Post>("Post");
```
Comment (pleerock) [inline test/functional/driver/convert-to-entity/convert-to-entity.ts]: `userRepository` should be `postRepository`

Comment (pleerock) [inline test/functional/driver/convert-to-entity/convert-to-entity.ts]: also no need to do `connection.getRepository<Post>("Post")` you can simply do `connection.getRepository(Post)`

File: test/functional/driver/convert-to-entity/convert-to-entity.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        let post = new Post();
+        post.id = 1;
+
+        await userRepository.save(post);
+
+        let loadedPost = await userRepository.findOneById(1);
+        if (loadedPost) {
+            expect(loadedPost.isNew).to.be.equal(null);
+        }
+    })));
```
Comment (pleerock) [inline test/functional/driver/convert-to-entity/convert-to-entity.ts]: not enough test cases. Save `isNew` with set to true, make sure its true, save `isNew` with false, make sure its equal to false as well, to make sure null functionality didn't break anything

File: test/functional/driver/convert-to-entity/convert-to-entity.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    after(() => closeTestingConnections(connections));
+
+    it.only("should return null value in entity property when record column is null", () => Promise.all(connections.map(async connection => {
+        const userRepository = connection.getRepository<Post>("Post");
+        let post = new Post();
+        post.id = 1;
+
+        await userRepository.save(post);
+
+        let loadedPost = await userRepository.findOneById(1);
```
Comment (pleerock) [inline test/functional/driver/convert-to-entity/convert-to-entity.ts]: `const`

File: test/functional/driver/convert-to-entity/convert-to-entity.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        entities: [Post],
+        schemaCreate: true,
+        dropSchemaOnConnection: true
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it.only("should return null value in entity property when record column is null", () => Promise.all(connections.map(async connection => {
+        const userRepository = connection.getRepository<Post>("Post");
+        let post = new Post();
```
Comment (pleerock) [inline test/functional/driver/convert-to-entity/convert-to-entity.ts]: `const`

**Comment by eyalhakim** [comment]
@pldin601 fixed, seems that again a test which is unrelated is failing

**Comment by pleerock** [comment]
Thank you for this fix! I'll publish a new version with this change tomorrow

**Comment by eyalhakim** [comment]
perfect, thank you very much!

**Comment by pleerock** [comment]
released in 0.1.0-alpha.27

**Pull request merged**